blocks/text.js

```diff
@@ -749,35 +750,138 @@ Blockly.Constants.Text.TEXT_JOIN_MUTATOR_MIXIN = {
           itemBlock.nextConnection.targetBlock();
     }
   },
+  storeValueConnections_: function() {
+    this.valueConnections_ = [];
+    for (var i = 0; i < this.itemCount_; i++) {
+      this.valueConnections_.push(this.getInput('ADD' + i).connection.targetConnection);
+    }
+  },
+  restoreValueConnections_: function() {
+    for (var i = 0; i < this.itemCount_; i++) {
+      Blockly.Mutator.reconnect(this.valueConnections_[i], this, 'ADD' + i);
+    }
+  },
+  addItem_: function() {
+    this.storeValueConnections_();
+    var update = function() {
+      this.itemCount_++;
+    };
+    this.update_(update);
+    this.restoreValueConnections_();
+    // Add shadow block
+    if (this.itemCount_ > 1) {
+      // Find shadow type
+      var firstInput = this.getInput('ADD' + 0);
+      if (firstInput && firstInput.connection.targetConnection) {
+        // Create a new shadow DOM with the same type as the first input
+        // but with an empty default value
+        var newInput = this.getInput('ADD' + (this.itemCount_ - 1));
+        var shadowInputDom = firstInput.connection.getShadowDom();
+        if (shadowInputDom) {
+          var shadowDom = document.createElement('shadow');
+          var shadowInputType = shadowInputDom.getAttribute('type');
+          shadowDom.setAttribute('type', shadowInputType);
+          var shadowDomField = document.createElement('field');
```

Comment:
Not sure why this field is being added. Looks like the field for a math_number block but the block type is coming from the shadow dom. Also, I think creating the field and not setting the content doesn't do anything